backend/src/tests/tests.test.js

```diff
@@ -1,117 +1,65 @@
-const contextPipeline = () => context({ req: reqMock, res: resMock });
-
-const server = new ApolloServer({
-    schema,
-    context: contextPipeline,
-    dataSources: () => {
-        return {
-          usersApi: usersApi,
-          postsApi: postsApi,
-          authApi: authApi
-        }
-    }
-});
-
-let mike_auth_token = '';
-let atanas_auth_token = '';
-
-const { query, mutate } = createTestClient(server);
+const { gql } = require("apollo-server");
+const {beforeEach, afterAll, describe, expect, it} = require("@jest/globals");
+const authService = require('../services/tokens');
+const actualContext = require('../context');
+const User = require('../domain/User')
+const Post = require('../domain/Post')
+const Server = require('../server');
+
+let query;
+let mutate;
+let requestMock;
+const context = () => actualContext({req: requestMock, authService});
+
+const cleanDatabase = async () => {
+    const { driver } = context();
+    await driver.session().writeTransaction(txc => txc.run("MATCH(n) DETACH DELETE n;"));
+}
 
 beforeEach(async () => {
-    process.env.JWT_SECRET='TEST_SECRET';
-    // For me a mock express.Request containing some headers is enough.
-    // I assume typescript will complain here:
-    reqMock = { headers: {} };
-    resMock = {};
-
-    let userIds = [uuidv4(), uuidv4()];
-
-    mike_auth_token = await authApi.createToken(userIds[0]);
-    atanas_auth_token = await authApi.createToken(userIds[1]);
-
-    postsApi.posts = [
-        { title: "Mike's Post 1", votes: 0, upvoters:[], author: userIds[0] },
-        { title: "Mike's Post 2", votes: 0, upvoters:[], author: userIds[0] },
-        { title: "Atanas's Post 1", votes: 0, upvoters:[], author: userIds[1] },
-        { title: "Atanas's Post 2", votes: 0, upvoters:[], author: userIds[1] },
-    ];
-
-    usersApi.users = [
-        {
-            id: userIds[0],
-            name: 'Mike',
-            posts: postsApi.posts.slice(0, 2)
-        },
-        {
-            id: userIds[1],
-            name: 'Atanas',
-            posts: postsApi.posts.slice(2, 4)
-        },
-    ];
-
-});
-
-describe("query", () => {
-    
-    it("posts with nested user object", async () => {
-        const GET_POSTS = gql`
-            query {
-                posts {
-                    title 
-                    author {
-                        name
-                    }
-                }
-            }
-        `;
-
-        const {
-            data: { posts }
-        } = await query({ query: GET_POSTS });
-
-        expect(posts).toHaveLength(4);
-        expect(posts).toEqual([
-            { title: "Mike's Post 1", author: { name: 'Mike' } },
-            { title: "Mike's Post 2", author: { name: 'Mike' } },
-            { title: "Atanas's Post 1", author: { name: 'Atanas' } },
-            { title: "Atanas's Post 2", author: { name: 'Atanas' } },
-        ]);
-    });
+    requestMock = { headers: {} };
+    await cleanDatabase();
```

Comment:
I like how you clean up your db before AND after transactions.
:star: :star: - Your software tests are free of side effects.